core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -202,6 +208,11 @@ class LogManager(val logDirs: Array[File],
                          delay = InitialTaskDelayMs,
                          period = flushCheckpointMs,
                          TimeUnit.MILLISECONDS)
+      scheduler.schedule("kafka-delete-topic",
+                         deleteLogs,
+                         delay = InitialTaskDelayMs,
```

Comment:
Potential issue in large or even moderate delay (e.g., a minute). When you do operations such as delete-topic or reassign-partitions on a large number of partitions it is possible that a large batch of partitions may need to be deleted in the same run of the scheduler. This can cause heavy IO and in some cases (due to that IO) GC as well on the broker. Do you see the need for throttling the deletion itself? Can you try the following: accumulate several gigs and set the deletion delay to be extremely high and delete several topic-partitions on one broker.